internal/backend/state_actions.go

```diff
@@ -46,91 +79,132 @@ func (state *State) actionUpdateMailbox(ctx context.Context, tx *ent.Tx, mboxID,
 	return state.apply(ctx, tx, imap.NewMailboxUpdated(mboxID, strings.Split(newName, state.delimiter)))
 }
 
-func (state *State) actionCreateMessage(ctx context.Context, tx *ent.Tx, mboxID string, literal []byte, flags imap.FlagSet, date time.Time) (int, error) {
-	res, err := state.remote.CreateMessage(state.metadataID, mboxID, literal, flags, date)
+func (state *State) actionCreateMessage(ctx context.Context, tx *ent.Tx, mboxID MailboxIDPair, literal []byte, flags imap.FlagSet, date time.Time) (int, error) {
+	internalID, res, err := state.remote.CreateMessage(ctx, state.metadataID, mboxID.RemoteID, literal, flags, date)
 	if err != nil {
 		return 0, err
 	}
 
 	update := imap.NewMessagesCreated()
 
-	if err := update.Add(res, literal, []string{mboxID}); err != nil {
+	if err := update.Add(res, literal, []imap.LabelID{mboxID.RemoteID}); err != nil {
```

Comment:
```suggestion
	if err := update.Add(res, literal, mboxID.RemoteID); err != nil {
```

i feel like `update.Add` should take the label IDs as a variadic parameter, would be nicer than casting to `[]imap.LabelID`

Reply:
Done